src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -63,6 +63,49 @@ public void Kill()
             }
         }
 
+        /// <summary>Stops the associated process immediately.</summary>
+        private void Stop()
+        {
+            EnsureState(State.HaveNonExitedId);
+            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGSTOP) != 0)
```

Comment:
What happens if this sends SIGSTOP to itself, i.e. if someone did `Process.GetCurrentProcess().Kill(true)`?

Reply:
It will stop and leave itself in a suspended state. :-( I'm going to work on addressing this. :-)